### PR TITLE
taler-exchange, taler-merchant: init at 0.8.1, 0.8.0

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

https://taler.net/

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/p2p/gnunet/default.nix [hunk 1]
```diff
@@ -1,7 +1,9 @@
 { lib, stdenv, fetchurl, adns, curl, gettext, gmp, gnutls, libextractor
 , libgcrypt, libgnurl, libidn, libmicrohttpd, libtool, libunistring
 , makeWrapper, ncurses, pkg-config, libxml2, sqlite, zlib
-, libpulseaudio, libopus, libogg, jansson, libsodium }:
+, libpulseaudio, libopus, libogg, jansson, libsodium
+
+, postgresqlSupport ? false, postgresql }:
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/p2p/gnunet/default.nix:6]:
```suggestion
, libpulseaudio, libopus, libogg, jansson, libsodium
, postgresqlSupport ? false, postgresql }:
```

File: pkgs/applications/networking/p2p/gnunet/default.nix [hunk 2]
```diff
@@ -19,7 +21,7 @@ stdenv.mkDerivation rec {
     adns curl gmp gnutls libextractor libgcrypt libgnurl libidn
     libmicrohttpd libunistring libxml2 ncurses gettext libsodium
     sqlite zlib libpulseaudio libopus libogg jansson
-  ];
+  ] ++ lib.optional (postgresqlSupport) postgresql;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/p2p/gnunet/default.nix:24]:
```suggestion
  ] ++ lib.optional postgresqlSupport postgresql;
```

File: pkgs/applications/networking/taler/default.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{ lib, stdenv, fetchurl, curl, gnunet, jansson, libgcrypt, libmicrohttpd
+, libqrencode, libsodium, libtool, pkg-config, postgresql, sqlite }:
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/taler/default.nix:2]:
```suggestion
, qrencode, libsodium, libtool, pkg-config, postgresql, sqlite }:
```

File: pkgs/applications/networking/taler/default.nix [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    patches = [ ./exchange-fix-6665.patch ];
+  };
+
+  taler-merchant = mkTaler {
+    pname = "taler-merchant";
+    version = "0.8.0";
+    sha256 = "sha256-scrFLXeoQirGqhc+bSQKRl84PfUvjrp1uxF7pfOIB9Q=";
+  } {
+    nativeBuildInputs = [ pkg-config ];
+    buildInputs = taler-exchange.buildInputs ++ [ libqrencode taler-exchange ];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/taler/default.nix:52]:
```suggestion
    buildInputs = taler-exchange.buildInputs ++ [ qrencode taler-exchange ];
```

File: pkgs/applications/networking/taler/default.nix [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  mkTaler = { pname, version, sha256 }:
+    extraAttrs:
+    stdenv.mkDerivation (extraAttrs // {
+      inherit pname version;
+      src = fetchurl {
+        url = "http://mirror.rit.edu/gnu/taler/${pname}-${version}.tar.gz";
+        inherit sha256;
+      };
+      enableParallelBuilding = true;
+      meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/taler/default.nix:16]:
```suggestion
      inherit pname version;

      src = fetchurl {
        url = "http://mirror.rit.edu/gnu/taler/${pname}-${version}.tar.gz";
        inherit sha256;
      };

      enableParallelBuilding = true;

      meta = with lib; {
```

File: pkgs/applications/networking/taler/default.nix [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    pname = "taler-merchant";
+    version = "0.8.0";
+    sha256 = "sha256-scrFLXeoQirGqhc+bSQKRl84PfUvjrp1uxF7pfOIB9Q=";
+  } {
+    nativeBuildInputs = [ pkg-config ];
+    buildInputs = taler-exchange.buildInputs ++ [ libqrencode taler-exchange ];
+    propagatedBuildInputs = [ gnunet' ];
+    PKG_CONFIG = "${pkg-config}/bin/pkg-config";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/taler/default.nix:56]:
```suggestion
  };
```

**Comment by Profpatsch** [comment]
yay, into blockchain-less future

**Pull request merged**